src/hazelcore/Statics_Exp.re

```diff
@@ -1203,19 +1329,22 @@ and ana_fix_holes_operand =
       };
     }
   | Case(_, scrut, rules) =>
-    let (scrut, scrut_ty, u_gen) =
+    let (scrut, ty_scrut, u_gen) =
       syn_fix_holes(ctx, u_gen, ~renumber_empty_holes, scrut);
     let (rules, u_gen) =
-      ana_fix_holes_rules(
-        ctx,
-        u_gen,
-        ~renumber_empty_holes,
-        rules,
-        scrut_ty,
-        ty,
-      );
-    (Case(StandardErrStatus(NotInHole), scrut, rules), u_gen);
-  }
+      ana_fix_holes_rules(ctx, u_gen, rules, ty_scrut, ty);
+    switch (syn_rule_types(ctx, ty_scrut, rules)) {
+    | None => failwith("ana_fix_holes_operand' Case 2")
+    | Some(clause_types) =>
+      switch (HTyp.join_all(GLB, clause_types)) {
```

Comment:
are we using LUB or GLB? seems inconsistent with joined_pattern_type function at the top of the file?